### ISSUE TITLE
rc2: client crash when opening settings -> interface tab on worlds with coordinate/minimap overlay disabled

### ISSUE THREAD

**Issue by wasakoki** [anegostudios/VintageStory-Issues#2062]
**Game Version:** 1.17.0-rc2
**Platform:** Windows
**Modded:** No
**SP/MP:** Singleplayer + Multiplayer   

### Description
client crashes when opening settings -> interface tab on worlds without coordinate/minimap overlay (AKA wilderness survival worlds)

### How to reproduce
1. create or join a world with coordinates/minimap setting disabled
2. open settings window, click on interface tab
3. game crashes

### Expected behavior
no crash

### Screenshots
n/a

### Logs
```
Running on 64 bit Windows with 16 GB RAM 
Game Version: v1.17.0-rc.2 (Unstable)
Loaded Mods: game@1.17.0-rc.2, creative@1.17.0-rc.2, survival@1.17.0-rc.2
28-Jul-22 16:24:21: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.Client.NoObf.GuiCompositeSettings.OnInterfaceOptions(Boolean on) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\MainMenu\Composite\GuiCompositeSettings.cs:line 1312
   at Vintagestory.API.Client.GuiElementToggleButton.OnMouseDownOnElement(ICoreClientAPI api, MouseEvent args) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryApi\Client\UI\Elements\Impl\Interactive\Controls\GuiElementToggleButton.cs:line 196
   at Vintagestory.API.Client.GuiElement.OnMouseDown(ICoreClientAPI api, MouseEvent mouse) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryApi\Client\UI\Elements\Impl\GuiElement.cs:line 638
   at Vintagestory.API.Client.GuiComposer.OnMouseDown(MouseEvent mouseArgs) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryApi\Client\UI\GuiComposer.cs:line 459
   at Vintagestory.API.Client.GuiDialog.OnMouseDown(MouseEvent args) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryApi\Client\UI\Dialog\GuiDialog.cs:line 536
   at Vintagestory.Client.NoObf.GuiDialogEscapeMenu.OnMouseDown(MouseEvent args) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\Systems\Gui\Dialogs\GuiDialogEscapeMenu.cs:line 380
   at Vintagestory.Client.NoObf.GuiManager.OnMouseDown(MouseEvent args) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\Systems\Gui\GuiManager.cs:line 394
   at Vintagestory.Client.NoObf.ClientPlatformWindows.Mouse_ButtonDown(Object sender, MouseButtonEventArgs e) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\ClientPlatform\Input.cs:line 220
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.Platform.Windows.WinGLNative.WindowProcedure(IntPtr handle, WindowMessage message, IntPtr wParam, IntPtr lParam) in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\Platform\Windows\WinGLNative.cs:line 792
   at OpenTK.Platform.Windows.Functions.DispatchMessage(MSG& msg)
   at OpenTK.Platform.Windows.WinGLNative.ProcessEvents() in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\Platform\Windows\WinGLNative.cs:line 1554
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second) in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\GameWindow.cs:line 368
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs)
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 87
-------------------------------

Event Log entries containing Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 27-Jul-22 07:21:54, Site = , Source = Windows Error Reporting, Message = Fault bucket 2215816033327233554, type 4
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.17.0.0
P3: 62e01e15
P4: KERNELBASE.dll
P5: 10.0.19041.1741
P6: e9b4a91b
P7: c0020001
P8: 0000000000034fd9
P9: 
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERC7DC.tmp.dmp
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERCA8D.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERCAAD.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERCAAB.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERCAEA.tmp.txt

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_Vintagestory.exe_b46670ad5e9a6820cd2251751726a821f43d097_30d1a8e0_2d405796-a1b1-4abe-92f4-1eb350bfcc6e

Analysis symbol: 
Rechecking for solution: 0
Report Id: 2df99be0-adab-4324-9768-dc7e708bdadc
Report Status: 268435456
Hashed bucket: f4135357ac59ac7dcec028f14215ba12
Cab Guid: 0 }
--------------
{ TimeGenerated = 27-Jul-22 07:21:53, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.17.0.0, time stamp: 0x62e01e15
Faulting module name: KERNELBASE.dll, version: 10.0.19041.1741, time stamp: 0xe9b4a91b
Exception code: 0xc0020001
Fault offset: 0x0000000000034fd9
Faulting process id: 0x7784
Faulting application start time: 0x01d8a1b36fea3d50
Faulting application path: E:\installs SSD\Vintagestory\Vintagestory\Vintagestory.exe
Faulting module path: C:\WINDOWS\System32\KERNELBASE.dll
Report Id: 2df99be0-adab-4324-9768-dc7e708bdadc
Faulting package full name: 
Faulting package-relative application ID:  }
--------------
{ TimeGenerated = 27-Jul-22 07:21:52, Site = , Source = .NET Runtime, Message = Application: Vintagestory.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: exception code c0020001, exception address 00007FF96D884FD9
Stack:
   at System.Environment._Exit(Int32)
   at System.Environment._Exit(Int32)
   at System.Environment.Exit(Int32)
   at System.EventHandler`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].Invoke(System.Object, System.__Canon)
   at OpenTK.NativeWindow.OnClosedInternal(System.Object, System.EventArgs)
   at System.EventHandler`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].Invoke(System.Object, System.__Canon)
   at OpenTK.Platform.Windows.WinGLNative.HandleDestroy(IntPtr, OpenTK.Platform.Windows.WindowMessage, IntPtr, IntPtr)
   at OpenTK.Platform.Windows.WinGLNative.WindowProcedure(IntPtr, OpenTK.Platform.Windows.WindowMessage, IntPtr, IntPtr)
   at OpenTK.Platform.Windows.Functions.DestroyWindow(IntPtr)
   at OpenTK.Platform.Windows.Functions.DestroyWindow(IntPtr)
   at OpenTK.Platform.Windows.WinGLNative.DestroyWindow()
   at OpenTK.Platform.Windows.WinGLNative.WindowProcedure(IntPtr, OpenTK.Platform.Windows.WindowMessage, IntPtr, IntPtr)
   at OpenTK.Platform.Windows.Functions.DefWindowProc(IntPtr, OpenTK.Platform.Windows.WindowMessage, IntPtr, IntPtr)
   at OpenTK.Platform.Windows.Functions.DefWindowProc(IntPtr, OpenTK.Platform.Windows.WindowMessage, IntPtr, IntPtr)
   at OpenTK.Platform.Windows.WinGLNative.WindowProcedure(IntPtr, OpenTK.Platform.Windows.WindowMessage, IntPtr, IntPtr)
   at OpenTK.Platform.Windows.Functions.PeekMessage(OpenTK.Platform.Windows.MSG ByRef, IntPtr, Int32, Int32, OpenTK.Platform.Windows.PeekMessageFlags)
   at OpenTK.Platform.Windows.Functions.PeekMessage(OpenTK.Platform.Windows.MSG ByRef, IntPtr, Int32, Int32, OpenTK.Platform.Windows.PeekMessageFlags)
   at OpenTK.Platform.Windows.WinGLNative.ProcessEvents()
   at OpenTK.GameWindow.Run(Double, Double)
   at Vintagestory.Client.ClientProgram.Start(Vintagestory.Client.ClientProgramArgs, System.String[])
   at Vintagestory.ClientNative.CrashReporter.Start(System.Threading.ThreadStart)
   at Vintagestory.Client.ClientProgram.Main(System.String[])
 }
```

**Comment by Thurasiz-thorn** [comment]
Still happening in rc3 Logs are in a different box I'll update with them later tonight.

**Comment by wasakoki** [comment]
still present in the stable release

**Comment by wasakoki** [comment]
fixed in 1.17.2
